pmdarima/arima/auto.py

```diff
@@ -173,6 +174,44 @@ def update(self, y, exogenous=None, maxiter=None, **kwargs):
     # TODO: decorator to automate all this composition + AIC, etc.
 
 
+class StepwiseContext(_context):
+    """Context Manager to capture runtime context for Stepwise mode.
+
+    ``StepwiseContext`` allows to call ``auto_arima`` in the context
+    of a runtime configuration that offers additional level of
+    control required in certain scenarios. Use cases that are either
+    sensitive to duration and/or the number of attempts to
+    find the best fit can use ``StepwiseContext`` to control them.
+    .
+
+    Parameters
+    ----------
+    max_steps : int, optional (default=100)
+        The maximum number of steps to try to find a best fit. When
+        the number of tries exceed this number, the stepwiese process
+        will stop the best fit model at that time will be returned.
+
+    max_dur : int, optional (default=None)
+        The maximum duration in seconds to try to find a best fit.
+        When the cumulative fit duration exceeds this number, the
+        stepwiese process will stop and the best fit model at that
+        time will be returned. Please note that this is a soft limit.
+    """
+
+    def __init__(self, max_steps=100, max_dur=None):
+        if max_steps is None or not 0 < max_steps <= 100:
+            raise ValueError('max_steps should be between 1 and 100')
```

Comment:
I wonder whether it makes sense to allow the user to specify more than 100? The default currently is 100 in the stepwise wrapper, but I like the pythonic contract of `I'm handing you the gun but trusting you not to shoot yourself in the foot`. If the user wants 1000, should we prevent that?